ext/src/ruby_api/linker.rs

```diff
@@ -32,28 +34,52 @@ impl DataTypeFunctions for Linker {
 }
 
 impl Linker {
+    /// @yard
+    /// @def new(engine)
+    /// @param engine [Engine]
+    /// @return [Linker]
     pub fn new(engine: &Engine) -> Result<Self, Error> {
         Ok(Self {
             inner: RefCell::new(LinkerImpl::new(engine.get())),
             refs: Default::default(),
         })
     }
 
+    /// @yard
+    /// Allow shadowing.
+    /// @def allow_shadowing=(val)
+    /// @param val [Boolean]
     pub fn set_allow_shadowing(&self, val: bool) {
         self.inner.borrow_mut().allow_shadowing(val);
     }
 
+    /// @yard
+    /// Allow unknown exports.
+    /// @def allow_unknown_exports=(val)
+    /// @param val [Boolean]
     pub fn set_allow_unknown_exports(&self, val: bool) {
         self.inner.borrow_mut().allow_unknown_exports(val);
     }
 
+    /// @yard
+    /// Define unknown (unresolved) imports as functions which trap.
+    /// @def define_unknown_imports_as_traps(mod)
+    /// @param mod [Module]
+    /// @return [void]
     pub fn define_unknown_imports_as_traps(&self, module: &Module) -> Result<(), Error> {
         self.inner
             .borrow_mut()
             .define_unknown_imports_as_traps(module.get())
             .map_err(|e| error!("{}", e))
     }
 
+    /// @yard
+    /// Define an item in this linker.
+    /// @def define(mod, name, item)
+    /// @param mod [String] Module name
+    /// @param name [Module] Import name
```

Comment:
Is this an instance of `Module`? Judging by the source it seems to be a string (I've never used Rust though)

(ditto for other methods in this class)

Reply:
Totally. Just fixed it, thanks!